drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -820,10 +861,41 @@ static int spi_mcux_init(const struct device *dev)
 	}
 
 	spi_context_unlock_unconditionally(&data->ctx);
+	pm_policy_device_power_lock_put(dev);
+
+	return 0;
+}
 
+static int spi_mcux_flexcomm_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		break;
+	case PM_DEVICE_ACTION_TURN_OFF:
+	    /*This flag is used to prevent configuration optimiztions
+	     * after exiting PM3 on spi_mcux_configure()
+	     */
+		reconfig = 1;
+		return 0;
```

Comment:
suggest changing this also to `break`, right now we are having 3 redundant `return 0` statements